docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,337 @@
+    "\n",
+    "```console\n",
+    "$ pip install onnx onnxmltools onnxruntime spox\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Usage\n",
+    "\n",
+    "```{warning}\n",
+    "It is important to notice that this method only works for {class}`~metalearners.TLearner`,\n",
+    "{class}`~metalearners.XLearner`, {class}`~metalearners.RLearner` and {class}`~metalearners.DRLearner`.\n",
+    "Converting an {class}`~metalearners.SLearner` is highly dependent on the fact that the base\n",
+    "model supports categorical variables or not and it is not implemented yet. \n",
+    "```\n",
+    "\n",
+    "### Loading the data\n",
+    "\n",
+    "Just like in our {ref}`example on estimating CATEs with a MetaLearner <example-basic>`,\n",
+    "we will first load some experiment data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from pathlib import Path\n",
+    "from git_root import git_root\n",
+    "\n",
+    "df = pd.read_csv(git_root(\"data/learning_mindset.zip\"))\n",
+    "outcome_column = \"achievement_score\"\n",
+    "treatment_column = \"intervention\"\n",
+    "feature_columns = [\n",
+    "    column for column in df.columns if column not in [outcome_column, treatment_column]\n",
+    "]\n",
+    "categorical_feature_columns = [\n",
+    "    \"ethnicity\",\n",
+    "    \"gender\",\n",
+    "    \"frst_in_family\",\n",
+    "    \"school_urbanicity\",\n",
+    "    \"schoolid\",\n",
+    "]\n",
+    "# Note that explicitly setting the dtype of these features to category\n",
+    "# allows both lightgbm as well as shap plots to\n",
+    "# 1. Operate on features which are not of type int, bool or float\n",
+    "# 2. Correctly interpret categoricals with int values to be\n",
+    "#    interpreted as categoricals, as compared to ordinals/numericals.\n",
+    "for categorical_feature_column in categorical_feature_columns:\n",
+    "    df[categorical_feature_column] = df[categorical_feature_column].astype(\"category\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we've loaded the experiment data, we can train a MetaLearner.\n",
+    "\n",
+    "\n",
+    "### Training a MetaLearner\n",
+    "\n",
+    "Again, mirroring our {ref}`example on estimating CATEs with a MetaLearner\n",
+    "<example-basic>`, we can train an\n",
+    "{class}`~metalearners.XLearner` as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from metalearners import XLearner\n",
+    "from lightgbm import LGBMRegressor, LGBMClassifier\n",
+    "\n",
+    "xlearner = XLearner(\n",
+    "    nuisance_model_factory=LGBMRegressor,\n",
+    "    propensity_model_factory=LGBMClassifier,\n",
+    "    treatment_model_factory=LGBMRegressor,\n",
```

Comment:
Are these usually LGBM estimators?

Reply:
No, but LGBM are the only ones that we can use the categorical features as real categoricals, we just have to convert the values to the codes before passing it to onnx, most `sklearn` models don't support native categoricals or the converters to onnx don't support them.

Reply:
Might be worth spelling out more explicitly